sdk/metric/doc.go

```diff
@@ -13,48 +13,157 @@
+Aggregators may be lock-free or they may use locking, but they should
+expect to be called concurrently.  Because of the tolerated race condition
+described above, aggregators must be capable of merging with another
+aggregator of the same type.
+
+Export Pipeline
+
+While the SDK serves to maintain a current set of records and
+coordinate collection, the behavior of a metrics export pipeline is
+configured through the export types in
+go.opentelemetry.io/otel/sdk/export/metric.  It is important to keep
+in mind the context these interfaces are called from.  There are two
+contexts, instrumentation context, where a user-level goroutine that
+enters the SDK resulting in a new record, and collection context,
+where a system-level thread performs a collection pass through the
+SDK.
+
+Descriptor is a struct that describes the metric instrument to the
+export pipeline, containing the name, recommended aggregation keys,
+units, description, metric kind (counter, gauge, or measure), number
+kind (int64 or float64), and whether the instrument has alternate
+semantics or not (i.e., monotonic=false counter, monotonic=true gauge,
+absolute=false measure).  A Descriptor accompanies metric data as it
+passes through the export pipeline.
+
+The AggregationSelector interface supports choosing the method of
+aggregation to apply to a particular instrument.  Given the
+Descriptor, this AggregatorFor method returns an implementation of
+Aggregator.  If this interface returns nil, the metric will be
+disabled.  The aggregator should be matched to the capabilities of the
+exporter.  Selecting the aggregator for counter and gauge instruments
+is relatively straightforward, but for measure instruments there are
+numerous choices with different cost and quality tradeoffs.
+
+Aggregator is an interface which implements a concrete strategy for
+aggregating metric updates.  Several Aggregator implementations are
+provided by the SDK.  Aggregators may be lock-free or use locking,
+depending on their structure and semantics.  Aggregators implement an
+Update method, called in instrumentation context, to receive a single
+metric event.  Aggregators implement a Checkpoint method, called in
+collection context, to save a checkpoint of the current state.
+Aggregators implement a Merge method, also called in collection
+context, that combines state from two aggregators into one.  Each SDK
+record has an associated aggregator.
+
+Batcher is an interface which sits between the SDK and an exporter.
+The Batcher embeds an AggregationSelector, used by the SDK to assign
+new Aggregators.  The Batcher supports a Process() API for submitting
+checkpointed aggregators to the batcher, and a ReadCheckpoint() API
+for producing a complete checkpoint for the exporter.  Two default
+Batcher implementations are provided, the "defaultkeys" Batcher groups
+aggregate metrics by their recommended Descriptor.Keys(), the
+"ungrouped" Batcher aggregates metrics at full dimensionality.
```

Comment:
How will aggregation scheme per metric will work? Will there be another Batcher that reads a config file or fetches it from a server and does the aggregation accordingly? OR exporters will have to implement their own aggregation?

I am looking for how to achieve creating something equivalent of views in opencensus.

In OC it also allowed to create more than one view on the same measure. For example, latency measurement in GRPC plugin can be aggergated as Count and Distribution in two different views. Both views could have different set of label-keys.